src/common/libsubprocess/local.c

```diff
@@ -480,7 +480,16 @@ static void sigchld_cb (pid_t pid, int status, void *arg)
 
 static int create_process (flux_subprocess_t *p)
 {
-    if (!(p->flags & FLUX_SUBPROCESS_FLAGS_FORK_EXEC)
+    bool allow_posix_spawn = true;
+#ifdef __MACH__
+    // This should not be necessary, but posix_spawn on macos emits an error if
+    // you ask it to close a file descriptor that is already closed.  It will
+    // take some work and refactoring to fix.  Likely also worth working in
+    // the new addchdir_np command rather than skipping it for getcwd.
+    allow_posix_spawn = false;
+#endif
+    if (allow_posix_spawn
+        && !(p->flags & FLUX_SUBPROCESS_FLAGS_FORK_EXEC)
         && !p->hooks.pre_exec
         && !flux_cmd_getcwd (p->cmd))
         return create_process_spawn (p);
```

Comment:
It might be a little cleaner to just add to `flux_local_exec_ex()`
```c
#if __APPLE__
    flags |= FLUX_SUBPROCESS_FLAGS_FORK_EXEC;
#endif
```
with the block comment rather than introduce the runtime conditional.

Reply:
I like that idea, will do.